src/iomusxml.cpp

```diff
@@ -1035,6 +1041,36 @@ bool MusicXmlInput::ReadMusicXmlMeasure(
         m_tieStopStack.clear();
     }
 
+    // add clef changes that might occur just before a bar line and remove inserted clefs from stack
+    if (!m_ClefChangeStack.empty()) {
+        for (auto staff : *measure->GetChildren()) {
+            assert(staff);
+            for (auto layer : *staff->GetChildren()) {
+                assert(layer);
+                std::vector<musicxml::ClefChange>::iterator iter;
+                for (iter = m_ClefChangeStack.begin(); iter != m_ClefChangeStack.end(); iter++) {
+                    if (iter->m_measureNum == measureNum && iter->m_staff == staff
+                        && iter->m_scoreOnset == m_durTotal) {
+                        if (iter->isFirst) { // add clef when first in staff
+                            layer->AddChild(iter->m_clef);
+                            iter->isFirst = false;
+                        }
+                        else {
+                            Clef *sameasClef = new Clef(); // add clef with @sameas referring to original clef
+                            sameasClef->SetSameas(iter->m_clef->GetUuid().c_str());
+                            layer->AddChild(sameasClef);
+                        }
+                    }
+                }
+            }
+        }
+        std::vector<musicxml::ClefChange>::iterator iter;
+        for (iter = m_ClefChangeStack.begin(); iter != m_ClefChangeStack.end(); iter++) {
+            if (!iter->isFirst) // remove from stack, if already inserted
+                m_ClefChangeStack.erase(iter--);
```

Comment:
Can you double check that you do not have a memory leak here? Erasing the ClefChange does not delete the vrv::Clef. If you want to delete it because it is not used and was not added to a Layer, then you have to do
```
delete->iter->m_clef;
```
I don't think you need to erase the iter, so then a C++11 range loop would work too.

Reply:
Thanks for the quick review!

I removed that loop and report un-matched clef changes at the end, before I clear the entire stack. (Coming from Java: how do I check memory leaks in Xcode?)